### PR TITLE
Add flexibility in searching for items using find command

### LINKED ISSUE #80: Add more flexibility in searching for items using find command

### PR DESCRIPTION
**Pull request by nishanthelango**
Closes #80

### CODE REVIEW

File: src/main/java/duchess/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -20,10 +41,29 @@ public FindCommand(List<String> words) {
     public void execute(Store store, Ui ui, Storage storage) throws DuchessException {
         if (words.size() > 0) {
```
Comment (limsiying) [inline src/main/java/duchess/logic/commands/FindCommand.java]: Just to clarfy when will words.size() == 0? 😮 If I'm not wrong wouldn't an empty user input throw a duchessexception in the parser?

Comment (limsiying) [inline src/main/java/duchess/logic/commands/FindCommand.java]: Can move the duchessexception up to make the happy path prominent 😄

Comment (nishanthelango) [reply src/main/java/duchess/logic/commands/FindCommand.java]: Oh right yeah I'll change it!

File: src/main/java/duchess/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -20,10 +41,29 @@ public FindCommand(List<String> words) {
...
-                            .filter(task -> task.containsKeyword(searchTerm))
-                            .collect(Collectors.toList());
+            List<Task> filteredTasks;
+            /*
+            If search term is enclosed by double quotation marks,
+            search for exact matches.
+             */
+            if (searchTerm.charAt(0) == '"' && searchTerm.charAt(searchTerm.length() - 1) == '"') {
+                filteredTasks =
+                        store.getTaskList().stream()
```
Comment (muserr) [inline src/main/java/duchess/logic/commands/FindCommand.java]: May want to consider separating in this manner?
` filteredTasks = store.getTaskList()`
`.stream()`

File: src/main/java/duchess/logic/commands/FindCommand.java [hunk 3]
```diff
@@ -20,10 +41,29 @@ public FindCommand(List<String> words) {
...
+                                .collect(Collectors.toList());
+            /*
+            Search for task descriptions with longest common subsequence of length
+            equal to at least 1 less than the length of the search term.
+             */
+            } else {
+                filteredTasks =
+                        store.getTaskList().stream()
+                                .filter(task -> lcs(task.getDescription().replaceAll(" ", ""),
+                                        searchTerm.replaceAll(" ", ""))
```
Comment (muserr) [inline src/main/java/duchess/logic/commands/FindCommand.java]: Similar to above, the terminating comma can be placed in the next line? (line 63)

File: src/main/java/duchess/logic/commands/FindCommand.java (hunk 2 above)
Comment (muserr) [inline src/main/java/duchess/logic/commands/FindCommand.java]: May want to consider separating in this manner?
```
filteredTasks = store.getTaskList()
.stream()
```

File: src/main/java/duchess/logic/commands/FindCommand.java [hunk 4]
```diff
@@ -12,6 +12,27 @@
...
     private List<String> words;
 
+    /**
+     * Returns the length of the longest common subsequence of 2 Strings.
+     *
+     * @param a the first String
+     * @param b the second String
+     * @return the length of the longest common subsequence
+     */
+    private int lcs(String a, String b) {
```
Comment (muserr) [inline src/main/java/duchess/logic/commands/FindCommand.java]: May want to use a more descriptive method name for lcs.

File: src/main/java/duchess/logic/commands/FindCommand.java [hunk 5]
```diff
@@ -12,6 +12,27 @@
...
+        int[][] table = new int[a.length() + 1][b.length() + 1];
+        for (int i = 1; i < table.length; i++) {
+            for (int j = 1; j < table[0].length; j++) {
```
Comment (muserr) [inline src/main/java/duchess/logic/commands/FindCommand.java]: May I suggest using 0-based indexing if possible? It will make the code look nicer :)

**Pull request merged**